sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/db/DatabaseCoordinator.kt

```diff
@@ -68,15 +69,36 @@ internal class DatabaseCoordinator private constructor(context: Context) {
     }
 
     /**
-     * Returns the file of the Cache database that would correspond to the given alias
-     * and network attributes.
+     * Returns the root folder of the Cache database that would correspond to the given
+     * alias and network attributes.
+     *
+     * @param network the network associated with the data in the cache database.
+     * @param alias the alias to convert into a database path
+     *
+     * @return the Cache database folder
+     */
+    internal suspend fun cacheDbRoot(
+        network: ZcashNetwork,
+        alias: String
+    ): File {
+        return newDatabaseFilePointer(
+            network,
+            alias,
+            DB_FS_CACHE_NAME,
+            Files.getZcashNoBackupSubdirectory(applicationContext)
+        )
+    }
+
+    /**
+     * Returns the file of the legacy Cache database that would correspond to the given
+     * alias and network attributes.
      *
      * @param network the network associated with the data in the cache database.
      * @param alias the alias to convert into a database path
      *
      * @return the Cache database file
      */
-    internal suspend fun cacheDbFile(
+    internal suspend fun legacyCacheDbFile(
```

Comment:
I don't think this is correct.  I read this as returning the database cache root directory under the old storage path, which never existed.

Historically, we were moving the database file to a new path.  But going forward we need to delete the database file under either the old path or the new path since we're transitioning to a filesystem cache instead of a database cache.

Reply:
Should this be handled by sdk or rust?? ...also i'm not sure when should the deletion occur ...

Reply:
Alright, thanks for your comments. Implemented with this in mind:
- Deleted from both the older and the newer legacy locations
- All related db files deleted - rollback files included
- The deletion is run once we try to access the new store blocks on disk root directory
- The deletion check does not throw any exception in case of failure, we just log it in console and try it on the next time
- Related new test added too